config/routes.rb

```diff
@@ -89,6 +89,7 @@ def matches?(request)
 
     resources :claims, only: [:index, :show] do
       resources :tasks, only: [:index, :show, :create], param: :name, constraints: {name: %r{#{ClaimCheckingTasks::TASK_NAMES.join("|")}}}
+      resources :payroll_gender_tasks, only: [:create], name: "payroll_gender"
```

Comment:
Did you try renaming the param so it mirrors the override `tasks` controller more closely?

```suggestion
      resources :payroll_gender_tasks, only: [:create], param: :name
```